### DISCUSSION TITLE
Aafter upgrading to UMI4 js files stop rendered , got exports error, please check this file esm format

### DISCUSSION THREAD

**Discussion by kudenv** [umijs/umi#11964] [Q&A]
hi, Comunity who can help?  I am using files with js extension, not a jsx, and out of the box umi 4 throws an error during the build process.
Parse file (all fies with js end)*ctions.js exports error, please check this file esm format.

my config.js file 
import { defineConfig } from "@umijs/max";
import {routes} from './routes';

export default defineConfig({
  plugins: ['@umijs/plugins/dist/antd', '@umijs/plugins/dist/locale'],  
  crossorigin: true,
  // alias
  alias: {
    '@': './src',
    '@@': './src/.umi',
    'worker': './public/worker',
    'assets': './public/assets',
    '__tests__': './__tests__',
  },
  esbuildMinifyIIFE: true,
  codeSplitting: {
    jsStrategy: 'granularChunks'
  },  
  routes,
  npmClient: 'yarn',
  fastRefresh: true,
  mfsu: true,
  two: {},
  initialState: {},
  module: {},
  layout:{ 
    title:'UmiJS', 
    locale:true, 
  },  
  antd : {

    compact: true,
    //babel-plugin-import
    import: true,
    // less or css, default less
    style: 'less',
    styleProvider : { 
     hashPriority: 'high',
     legacyTransformer: true,
   },
  }
});

package.json

 "scripts": {
    "dev": "max dev",
    "build": "max build",
    "postinstall": "max setup",
    "setup": "max setup",
    "start": "npm run dev"
  },
  "dependencies": {
    "babel-plugin-formatjs": "^10.5.10",
    "classnames": "^2.3.2",
    "react-icons": "^4.12.0",
    "react-responsive": "^9.0.2"
  },
  "devDependencies": {
    "@ant-design/icons": "^5.2.6",
    "@types/react": "^18.2.37",
    "@types/react-dom": "^18.2.15",
    "@umijs/max": "^4.0.88",
    "antd": "^5.11.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "mockjs": "^1.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "rimraf": "^5.0.5"
  },
  "lint-staged": {
    "*.{js,jsx,css,less}": [
      "max lint --fix",
      "git add"
    ]
  }

**Comment by fz6m** [comment]
Check reporting errors and make sure the files are esm format.

**Reply by kudenv** [reply]
my files in esm format, i want what files with js extensions written in esm format  been parsed as JSX files .

**Reply by fz6m** [reply]
If you want to write `React JSX element` , the file must end with `.jsx` / `.tsx` (not `.js`)

**Reply by MaazAhmadDeveloper** [reply]
thanks
